runner/scheduler.py

```diff
@@ -313,43 +285,49 @@ def initialise_result(self, example: TestExample) -> tuple[TestResult, bool]:
             result.load(self.agent.name)
         return result, skip
 
-    def run_interleaved_group(self, test_group: list[TestExample]):
-        if not self.config.continuous_conversation:
-            self.agent.reset()
-
+    def run_tests(self):
         self.result_callbacks = []
-        self.group_master_log = []
+        self.master_log = []
+        datasets_run = set()
         results = dict()
-        colour_print("Green", f"Group has {len(test_group)} tests.")
         finished = 0
-        for example in self.iter_tests(test_group):
+        for example in self.iter_tests(self.tests):
+            self.progress_dialog.notify_running(example)
+
             skip = False
             if example.unique_id not in results:
                 result, skip = self.initialise_result(example)
                 results[example.unique_id] = result
                 example.finished = skip
 
+                # Ask the agent to ignore any previous related information.
+                if not skip and example.reset_message != "" and type(example.dataset_generator) in datasets_run:
+                    action = SendMessageAction(example.reset_message)
+                    self.log_action(example, action)
+                    self.send_message(action)
+
             while not example.finished:
                 action = example.step()
                 if action is None:
                     break
                 self.log_action(example, action)  # Attributes are modified afterwards.
+                if isinstance(action, WaitAction):
+                    self.set_to_wait(example.unique_id, action)
+                    break
                 if isinstance(action, (SendMessageAction, SendAndRegisterAction)):
                     # TODO: the test should autonomously create the question
                     if example.is_temporal and action.is_question:
                         action.message = create_question(example, self.action_logs)
                     self.send_message(action)
                     if isinstance(action, SendAndRegisterAction):
                         self.register_callback(example, results[example.unique_id])
-                elif isinstance(action, WaitAction):
-                    self.set_to_wait(example.unique_id, action)
-                    break
 
             self.check_result_callbacks()
 
             if example.finished:
                 finished += 1
                 result = results[example.unique_id]
+                datasets_run.add(type(example.dataset_generator))
                 self.progress_dialog.notify_result(result)
                 if not skip:
                     reset_message = example.reset_message
```

Comment:
Below this line is a reference to `self.config.continuous_conversation` that  has not been removed

Reply:
That shouldn't be there. Let me check, I might have forgotten to push.